### PR TITLE
Modify user agent header syntax and add support for app ID

### PR DESCRIPTION
**Pull request by wty-Bryant**
Modify http request User-Agent header syntax and add support for optional app ID section in the UA header.
Customer code could offer an App ID for identifying their applications, which can be obtained from following sources in descending priority:
1. Client configuration
2. Environment variables with name of AWS_SDK_UA_APP_ID
3. Shared config profile field called sdk_ua_app_id

Once a service operation is called with app ID set, it will be concatenated to UA header string in format `{app/appID}`. One example UA string calling s3 service is shown here:
`User-Agent: aws-sdk-go-v2/1.18.1 os/macos lang/go#1.19.5 md/GOOS#darwin md/GOARCH#amd64 api/s3#1.36.0 app/123`

### CODE REVIEW

File: aws/middleware/user_agent.go
Comment (aajtodd) [inline aws/middleware/user_agent.go:82]: **fix**: `appId` is singular, also this is forcing us to know/care about appId when creating the user agent structure.

There is already a `SDKAgentKeyType` of type `ApplicationIdentifier` for this purpose. We should revert this back and just make an explicit call in setting up the middleware to add two keys. I'll expand elsewhere.

File: service/accessanalyzer/api_client.go
Comment (aajtodd) [inline service/accessanalyzer/api_client.go:362]: Rather than force AppId through this API and abuse variadic arguments we can just call it twice I believe.

e.g.

```go
if err := awsmiddleware.AddSDKAgentKeyValue(awsmiddleware.APIMetadata, "accessanalyzer", goModuleVersion)(stack); err != nil { 
    return err
}

if options.AppID != nil {
    return awsmiddleware.AddSDKAgentKey(awsmiddleware.ApplicationIdentifier, options.AppID)(stack)
}

return nil
```

**Review by aajtodd — CHANGES_REQUESTED** [review]

File: aws/config.go
Comment (aajtodd) [inline aws/config.go:136]: > AppId is an optional application specific identifier that can be set. When set it will be appended to the User-Agent header of every request in the form of `App/{AppId}`. This variable is sourced from environment variable `AWS_SDK_UA_APP_ID` or the shared config profile attribute `sdk_ua_app_id`. See https://docs.aws.amazon.com/sdkref/latest/guide/settings-reference.html for more information on environment variables and shared config settings.

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java
Comment (aajtodd) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java:208]: This doesn't match the current generated setting:
> The sdk user agent app id to be added to http request's User-Agent header
	AppID string


Maybe need to regenerate?

**suggestion**: 
> The (optional) application specific identifier appended to the  User-Agent header.

**Review by aajtodd — APPROVED** [review]
Approved, but wait for second review and green light to merge.

**Review by lucix-aws — APPROVED** [review]

**Pull request merged**
